src/components/welcome/MyAccounts/index.tsx

```diff
@@ -13,21 +13,32 @@ import AddIcon from '@/public/images/common/add.svg'
 import { AppRoutes } from '@/config/routes'
 import ConnectWalletButton from '@/components/common/ConnectWallet/ConnectWalletButton'
 import useWallet from '@/hooks/wallets/useWallet'
+import { useRouter } from 'next/router'
 
 const NO_SAFES_MESSAGE = "You don't have any Safe Accounts yet"
-import { useRouter } from 'next/router'
 
 type AccountsListProps = {
-  safes: SafeItems
+  safes: SafeItems | undefined
   onLinkClick?: () => void
 }
 const AccountsList = ({ safes, onLinkClick }: AccountsListProps) => {
-  const router = useRouter()
-  const ownedSafes = useMemo(() => safes.filter(({ isWatchlist }) => !isWatchlist), [safes])
-  const watchlistSafes = useMemo(() => safes.filter(({ isWatchlist }) => isWatchlist), [safes])
   const wallet = useWallet()
-  const trackingLabel =
-    router.pathname === AppRoutes.welcome.accounts ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+  const router = useRouter()
+
+  const ownedSafes = useMemo(() => safes?.filter(({ isWatchlist }) => !isWatchlist), [safes])
+  const watchlistSafes = useMemo(() => safes?.filter(({ isWatchlist }) => isWatchlist), [safes])
+
+  const isLoginPage = router.pathname === AppRoutes.welcome.accounts
+  const trackingLabel = isLoginPage ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+
+  useEffect(() => {
+    // track owned and watchlist safe counts once when the user logs in
+    if (watchlistSafes && ownedSafes && isLoginPage) {
+      trackEvent({ ...OVERVIEW_EVENTS.TOTAL_SAFES_OWNED, label: ownedSafes.length })
+      trackEvent({ ...OVERVIEW_EVENTS.TOTAL_SAFES_WATCHLIST, label: watchlistSafes.length })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isLoginPage])
```

Comment:
This won't work if safes aren't defined on the initial component render. You should keep all the dependencies but add a global flag to do it only once per page.

Besides, please extract that logic into a hook.